DESIGN.md

```diff
@@ -0,0 +1,126 @@
+No update functionality will be provided in terms of resources control - if the user wants to add or change some limits he is required to stop the job and create a new one with the required parameters.
+The following set of flags is used for this purpose, user can set the required limit in one of the groups:
+    * **mem** - memory limit for a job in megabytes
+    * **cpu** - cpu share in percents (1-100) available to this job
+    * **io** - proportion of I/O access (1-100) available to this job
+1. Stop the job: a user is required to provide a job ID for the job termination. The default behavior is to kill the task as it will trigger _SIGKILL_ to be sent for the command termination.
+1. Get the status of the job. Requires only the job ID to be sent, the user gets in return the job status, all the job resource limits set upon job creation and exit code (applies only if the job is in the finished or stopped status).
+It is guaranteed that the task will be terminated during the request.
+1. Stream the output of the job. Requires only the job ID, starts the stream of the job stdout - the user gets everything that was written by the command until that moment and continues to get the command logs in real time until either the job is finished/terminated or the user interrupts the stream command execution (CTRL-C).
+It is a completely valid scenario to request the logs of both stdout and stderr (or even stdout and once again stdout) of the same job at the same time. 
+    * **err** - optional flag, if provided starts the stream of stderr instead of stdout
+
+Some examples can be found under the Usage section of the readme file.
+
+
+## Security
+
+TLS of version 1.3 will be used as it will provide faster and more secure connections. Older versions won't be supported. 
+As the TLS 1.3 ciphersuites are not configurable the following default list will be used:
+```
+TLS_AES_128_GCM_SHA256
+TLS_AES_256_GCM_SHA384
+TLS_CHACHA20_POLY1305_SHA256
+```
+The CA certificate will be either presented in the repository or generated using the provided script.
+The same goes for the client and server keys and certificates, which will be signed using the mentioned CA and provided for local use.
+
+We will use ed25519 algorithm to generate the private keys, which later will be used along with the certificates for setting up the gRPC server and client connection respectively.
+mTLS authentication will be used. As we don't want to introduce some sophisticated auth system in this exercise we might consider all the users with valid certificates to be authorized within the system. 
+
+In order to identify the user we might extract CN and use it as a user login within the system. Later this login might be used to determine whether the user is capable of managing a task, i.e. all attempts to terminate or get status of the job created by another user will be declined.  
```

Comment:
> Later this login might be used to determine whether the user is capable of managing a task, i.e. all attempts to terminate or get status of the job created by another user will be declined.  

Later as the initial implementation will not support this?

Reply:
I'm going to implement this as it is not supposed to take much time. I've updated the wording so it'll be more clear